pages-domain/src/main/java/nl/vpro/domain/page/RelationDefinitionServiceProvider.java

```diff
@@ -0,0 +1,31 @@
+package nl.vpro.domain.page;
+
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * @author Michiel Meeuwissen
+ * @since 5.15.4
+ */
+@Slf4j
+public class RelationDefinitionServiceProvider {
+
+    private RelationDefinitionServiceProvider() {
+
+    }
+
+    private static RelationDefinitionService instance;
+
+    public static void setInstance(RelationDefinitionService instance) {
```

Comment:
This certainly is better the the old solution. Did you ever consider using a ServiceLoader for providers? This would not return a singleton of course but possibly multiple implementations.

Reply:
I use it sometimes, it may be a solution here, but in pages publisher it will be instantiated by spring, I don't know if that would be feasible here. 

Anyway it would not invalidate this, when using serviceloaders you still have to have some code to actually do it.